File: src/test/main/MainTest.java
package test.main;

import java.io.ByteArrayOutputStream;
import java.io.PrintStream;
import org.junit.AfterClass;
import org.junit.BeforeClass;
import org.junit.Test;
import org.junit.runner.RunWith;
import test.OrderedRunner;

@RunWith(OrderedRunner.class)
public class MainTest {

    private final PrintStream systemOut = System.out;
    private ByteArrayOutputStream testOut;

    @BeforeClass
    public static void setUpOutput() {
        /*testOut = new ByteArrayOutputStream();
        System.setOut(new PrintStream(testOut));*/
    }

    private String getOutput() {
        return testOut.toString();
    }

    @AfterClass
    public static void restoreSystemOutput() {
        //System.setOut(systemOut);
    }

    @Test
    public void shouldPrintOnTerminal() {
        //Main.main(null);
        //assertEquals("", getOutput());
    }
}

File: src/test/main/MainTest.kt
package test.main

import app.main.Main
import org.junit.After
import org.junit.Before
import org.junit.Test
import java.io.PrintStream
import kotlin.test.assertEquals
import java.io.ByteArrayOutputStream

class MainTest {
    private val systemOut: PrintStream = System.out
    private var testOut: ByteArrayOutputStream? = null

    @Before
    fun setUpOutput() {
        testOut = ByteArrayOutputStream()
        System.setOut(PrintStream(testOut))
    }

    fun getOutput(): String {
        return testOut.toString()
    }

    @After
    fun restoreSystemOutput() {
        System.setOut(systemOut)
    }

    @Test
    fun shouldPrintInTerminal() {
        Main.main(null)
        assertEquals("De locos\n", getOutput())
    }
}
